File: app/src/main/java/com/example/mytemplate/config/RetrofitConfig.java
package com.example.mytemplate.config;

import com.example.mytemplate.api.GlobalApiServices;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitConfig {

    private final static String baseUrl = GlobalConfig.getBaseUrlExample();

    private static Retrofit getRetrofitBuilder(String baseUrl){
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        httpClient.addInterceptor(logging);

        Retrofit.Builder builder = new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create());

        return builder
                .client(httpClient.build())
                .build();
    }

    public static GlobalApiServices clientService(){
        return getRetrofitBuilder(baseUrl)
                .create(GlobalApiServices.class);
    }

    public static GlobalApiServices clientService(String baseUrl){
        return getRetrofitBuilder(baseUrl)
                .create(GlobalApiServices.class);
    }
}

File: app/src/main/java/com/example/mytemplate/config/RetrofitConfig.kt
package com.example.mytemplate.config

import com.example.mytemplate.BuildConfig
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitConfig {
    @Suppress("SameParameterValue")
    private fun httpClientBuilder(isDebug: Boolean) = OkHttpClient.Builder().apply {
        if(isDebug)
            addInterceptor(HttpLoggingInterceptor().apply {
                level = HttpLoggingInterceptor.Level.BODY
            })
    }.build()

    fun getRetrofitBuilder(): Retrofit = Retrofit.Builder()
            .baseUrl(BuildConfig.SERVER_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(httpClientBuilder(GlobalConfig.IS_DEBUG))
            .build()
}